#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *


PACKAGE = "bachelors_final_project"

gen = ParameterGenerator()

gen.add("normalDistanceWeight", double_t, 0,
        "Set the relative weight (between 0 and 1) to give to the angular distance",
        .1, 0, 1)
gen.add("distanceThreshold", double_t, 0,
        "Points within this distance from the hypothesized plane are scored as inliers",
        .1, 0, 1)
gen.add("maxDepthChangeFactor", double_t, 0,
        "The depth change threshold for computing object borders in integral normal estimation. ",
        0.02, 0, 1)
gen.add("useDepthDependentSmoothing", bool_t, 0,
        "Set if using depth dependent smoothing",
        True)
gen.add("normalSmoothingSize", double_t, 0,
        "Factor which influences the size of the area used to smooth normals",
        10.0, 0)
gen.add("leafSize", double_t, 0,
        "The voxel grid leaf size.",
        .005, 0, 1)
gen.add("minHeight", double_t, 0,
        "Min prism height",
        0, 0, 1)
gen.add("maxHeight", double_t, 0,
        "Max prism height",
        0.2, 0, 1)

gen.add("defaultParams", bool_t, 0,
        "Return values to default",
        False)

exit(gen.generate(PACKAGE, PACKAGE, "Parameters"))
